lib/tlaw/dsl.rb

```diff
@@ -289,41 +298,61 @@ module DSL
     #     ...and so on. See also {#param} for understanding what you
     #     can change here.
 
-    # @!method post_process(key = nil, &block)
+    # @!method transform(key = nil, replace: false, &block)
     #   Sets post-processors for response.
     #
-    #   There are also {#post_process_replace} (for replacing entire
-    #   response with something else) and {#post_process_items} (for
-    #   post-processing each item of sub-array).
+    #   There are also the `replace` option (for replacing entire
+    #   response with something else), set to `false` by default.
+    #   Alternatively, {#transform_items} is a separate method for
+    #   post-processing each item of sub-array.
     #
     #   Notes:
     #
     #   * you can set any number of post-processors of any kind, and they
-    #     will be applied in exactly the same order they are set;
+    #     will be applied in exactly the same order they are defined;
     #   * you can set post-processors in parent namespace (or for entire
     #     API), in this case post-processors of _outer_ namespace are
-    #     always applied before inner ones. That allow you to define some
-    #     generic parsing/rewriting on API level, then more specific
-    #     key postprocessors on endpoints;
+    #     always applied before inner ones. That allows you to define some
+    #     generic parsing/rewriting at the API level, then more specific
+    #     key post-processors on endpoints;
     #   * hashes are flattened again after _each_ post-processor, so if
     #     for some `key` you'll return `{count: 1, continue: false}`,
-    #     response hash will immediately have
+    #     the response hash will immediately have
     #     `{"key.count" => 1, "key.continue" => false}`.
     #
-    #   @overload post_process(&block)
+    #   @overload transform(replace: false, &block)
     #     Sets post-processor for whole response. Note, that in this case
     #     _return_ value of block is ignored, it is expected that your
```

Comment:
This "Note,..." is not true anymore, considering `replace:` option :)

Reply:
With the cherry-picks, I'm confused about what ended up wrong with the moving things around to separate PRs. I'll have to look this over.

Reply:
OK, looking at this, it is correct. The new example illustrates the same effect for when `replace: false` is passed (which is also the default). Further down you'll see an example of what happens with `replace: true`, which is new documentation.

I think the note indicates that `transform(replace: false)` is a misnomer and also a bit confusing. My thought is that if you don't want to transform the value, this should be a method name like `handle` or something (e.g. maybe `process` for this specific behavior) where the return value is not expected to be processed. (I noted in another comment that I would prefer to make the default behavior of `transform` to be `replace: true`.)